src/veo_api.cpp

```diff
@@ -316,6 +316,64 @@ int veo_alloc_mem(veo_proc_handle *h, uint64_t *addr, const size_t size)
   return 0;
 }
 
+/*
+int veo_get_proc_identifier(veo_proc_handle *h)
+{
+  return ProcHandleFromC(h)->getProcIdentifier();
+}
+*/
+
+/**
+ * @brief Allocate a VE memory buffer
+ *
+ * This function returns the address with the process identifier. 
+ * Since the number of processes that can be identified by the process 
+ * identifier is 8, this function will fail if the user attempts to 
+ * identify more than 8 processes. The memory allocated using this 
+ * function must be transferred to the VE side with veo_args_set_hmem() 
+ * and freed with veo_free_hmem(). And the data transfer of this 
+ * memory between VH and VE must be done by veo_hmemcpy(). veo_hmemcpy() 
+ * allows user to transfer data from VH to VE, from VE to VH, or from 
+ * VH to VH. The transfer direction is determined from the identifier 
+ * of the virtual address of the argument passed to veo_hmemcpy(). 
+ * User can check with veo_is_ve_addr() that the target address is 
+ * flagged with the VE process identifier. So, if user pass this 
+ * memory to veo_is_ve_addr(), it will return 1.
+ *
+ * @param h VEO process handle
+ * @param addr [out] a pointer to VEMVA address with the identifier
+ * @param size [in] size in bytes
+ * @retval 0 memory allocation succeeded.
+ * @retval -1 memory allocation or proc identifier acquisition failed.
+ * @retval -2 internal error.
+ */
+int veo_alloc_hmem(veo_proc_handle *h, void **addr, const size_t size)
```

Comment:
Because we call this **heterogeneous memory** I believe that there should be a way to also allocate memory on the host with this call. Otherwise this function is more or less a "allocate on VEs" function. We can't overload the C API, therefore the only thing that comes to my mind is: allocate on VH when proc handle is nullptr. Not very pretty, but could you consider this?